source/Calamari.Common/Features/StructuredVariables/YamlFormatVariableReplacer.cs

```diff
@@ -103,13 +104,19 @@ public void ModifyFile(string filePath, IVariables variables)
                 }
 
                 // Write the replacement file
-                using (var writer = new StreamWriter(filePath, false, encoding))
-                {
-                    writer.NewLine = lineEnding == StringExtensions.LineEnding.Dos ? "\r\n" : "\n";
-                    var emitter = new Emitter(writer, indentDetector.GetMostCommonIndent());
-                    foreach (var outputEvent in outputEvents)
-                        emitter.Emit(outputEvent);
-                }
+                if (EncodingAdaptiveFileWriter.EncoderDoesNotRaiseErrorsForUnsupportedCharacters(encoding))
+                    encoding = Encoding.GetEncoding(encoding.WebName, EncoderFallback.ExceptionFallback, DecoderFallback.ExceptionFallback);
```

Comment:
Note: I'm about to move this check & fix inside `EncodingAdaptiveFileWriter`, and incorporate `EncodingAdaptiveFileWriter` into `ICalamariFileSystem`

Reply:
Done.